x/incentive/types/adapter.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+// SourceAdapter provides source shares from an external module.
+type SourceAdapter interface {
+	// GetShares returns source shares owned by one address.
+	//
+	// For example, the shares a user owns in the kava:usdx and bnb:usdx swap pools.
+	// It returns the shares for several sources at once, in the same order as the sourceIDs. Specifying no sourceIDS will return no shares.
+	GetShares(ctx sdk.Context, owner sdk.AccAddress, sourceIDs []string) []sdk.Dec
```

Comment:
Might be a good idea to prefix this -- helps for clarity when not being compared to TotalShares and scope code searches (grep, ctags, etc).

Maybe `GetOwnerShares` or `OwnerSharesBySource`, etc?

Reply:
I normally try and rely on the containing thing (`SourceAdapter` in this case) and argument types to add context to avoid realy long names, but these do seem clearer